### DISCUSSION TITLE
MOBILE ARM64 SUPPORT

### DISCUSSION THREAD

**Discussion by v8karlo** [mifi/lossless-cut#1054] [Support]
I have installed ubuntu 21.04 on my Samsung s20 phone and is working great.
Non root via Termux.
Through DEX have output to large screen and using Ubuntu on my phone as my laptop all the time.

Can you make release for ARM64 phone?

It will be great to have LossLessCut on my phone arm64 Ubuntu!

Thanks in advance!

**Comment by mifi** [comment]
I did check, but I don't think it's possible without a self hosted github actions runner. I'm not going to spend any time on setting up a self-hosted runner, but if someone wants to, then I could be up for adding support for it. Need to compile also ffmpeg/ffprobe for arm64 so it's not that trivial though.

https://github.community/t/ask-about-github-hosted-runners-arm64-platform-support/167483

In the mean time, is it possible to emulate x86 from arm64?

**Comment by v8karlo** [comment]
The ffmpeg for Raspberry Pi which is arm64 I saw here:

https://github.com/snapcrafters/ffmpeg

The arm64 is self growing market, more and more people install linux on their phones and use them on large screens.

I am not expert so I can not help a lot, but arm64 is future, and it will happen very fast.

My phone Samsung s20 can do everything as my laptop, maybe even more.

**Comment by v8karlo** [comment]
Also I found arm64 ffmpeg here:

https://blog.actorsfit.com/a?ID=01750-95282f95-f37d-4532-9327-0664754a9b14

I did not even search for very long, people already saw the importance of arm64 on mobile phones and similar.

**Comment by v8karlo** [comment]
I checked on my phone arm64 (installed ubuntu 20.04) for ffmpeg and it is in repository.

arm64

root@localhost:~# sudo apt install ffmpeg
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  libavdevice58 libavresample4
Suggested packages:
  ffmpeg-doc
The following NEW packages will be installed:
  ffmpeg libavdevice58 libavresample4
0 upgraded, 3 newly installed, 0 to remove and 0 not upgraded.
Need to get 1646 kB of archives.
After this operation, 2568 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://ports.ubuntu.com/ubuntu-ports hirsute/universe arm64 libavdevice58 arm64 7:4.3.2-0+deb11u1ubuntu1 [71.4 kB]
Get:2 http://ports.ubuntu.com/ubuntu-ports hirsute/universe arm64 libavresample4 arm64 7:4.3.2-0+deb11u1ubuntu1 [42.5 kB]
Get:3 http://ports.ubuntu.com/ubuntu-ports hirsute/universe arm64 ffmpeg arm64 7:4.3.2-0+deb11u1ubuntu1 [1532 kB]
Fetched 1646 kB in 2s (661 kB/s) 
Selecting previously unselected package libavdevice58:arm64.
(Reading database ... 153086 files and directories currently installed.)
Preparing to unpack .../libavdevice58_7%3a4.3.2-0+deb11u1ubuntu1_arm64.deb ...
Unpacking libavdevice58:arm64 (7:4.3.2-0+deb11u1ubuntu1) ...
Selecting previously unselected package libavresample4:arm64.
Preparing to unpack .../libavresample4_7%3a4.3.2-0+deb11u1ubuntu1_arm64.deb ...
Unpacking libavresample4:arm64 (7:4.3.2-0+deb11u1ubuntu1) ...
Selecting previously unselected package ffmpeg.
Preparing to unpack .../ffmpeg_7%3a4.3.2-0+deb11u1ubuntu1_arm64.deb ...
Unpacking ffmpeg (7:4.3.2-0+deb11u1ubuntu1) ...
Setting up libavresample4:arm64 (7:4.3.2-0+deb11u1ubuntu1) ...
Setting up libavdevice58:arm64 (7:4.3.2-0+deb11u1ubuntu1) ...
Setting up ffmpeg (7:4.3.2-0+deb11u1ubuntu1) ...
Processing triggers for bamfdaemon (0.5.5+21.04.20210119-0ubuntu1) ...
Rebuilding /usr/share/applications/bamf-2.index...
Processing triggers for desktop-file-utils (0.26-1ubuntu1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for gnome-menus (3.36.0-1ubuntu1) ...
Processing triggers for libc-bin (2.33-0ubuntu5) ...
Processing triggers for man-db (2.9.4-2) ...
Processing triggers for mailcap (3.68ubuntu1) ...
root@localhost:~#

**Comment by v8karlo** [comment]
To install Ubuntu on phone without root:

https://github.com/RandomCoderOrg/ubuntu-on-android

All you need is Termux and Bvnc free and in few steps Ubuntu is installed.

And you have arm64 phone full working Linux.

**Comment by mifi** [comment]
Thanks for your inpu. I created an issue from this discussion: https://github.com/mifi/lossless-cut/issues/1056

**Comment by v8karlo** [comment]
Thank you on your unique software.
I use it a lot on my PC, and lately I am using mobile on large screen wherever I go, 
and I am just missing losslesscut on my linux phone.

Thank you in advance for your efforts!

**Comment by v8karlo** [comment]
On screenshoot you can see android phone DEX (desktop mode) and Ubuntu in window. 
The screenshoot is from my Samsung S20 phone which I use instead of my laptop for some time now.

Combining Samsung DEX on android phone with no root Linux Ubuntu can turn your phone into powerfull small PC like station.
Used on large screen it is full desktop experience.
Arm64 on phones will become very fast popular platform.

Developing desktop apps for android is becomming a need and is whole new area.
Or for the arm64 linux which can be used as platform on phones.

Read more about it here:
https://github.com/RandomCoderOrg/ubuntu-on-android/discussions/126

![image](https://user-images.githubusercontent.com/57914925/157269835-0902d9fe-fb44-42a9-9431-d8dbcd716141.png)

![image](https://user-images.githubusercontent.com/57914925/157271291-34569706-fdb8-440c-98cb-543c3cdef2b2.png)

![image](https://user-images.githubusercontent.com/57914925/157271380-a8ff1a8d-e2ea-4a89-946f-9d5b158f33e0.png)

![image](https://user-images.githubusercontent.com/57914925/157271461-4f908ae2-63fa-49d1-bc46-b19d83f7fb1d.png)

![image](https://user-images.githubusercontent.com/57914925/157271554-f9cf26a0-227e-49b4-a0ee-636b0924dc23.png)

**Comment by mifi** [comment]
I tried to build for arm64 linux now. You can try it out. Note that you may get an error the first time you start up because there is no ffmpeg included. You can go to settings and specify the path to ffmpeg and ffprobe

**Comment by v8karlo** [comment]
I opened discussion on project pages of Arm64 mobile Linux here:

https://github.com/RandomCoderOrg/ubuntu-on-android/discussions/129

and asked for help in testing your build.

Thank you very much on your efforts !

..

**Comment by v8karlo** [comment]
I am novice in Linux world , but thanks to authors of Udroid Linux on mobile phones it worked like a charm !

Mr. Saicharan Kandukuri and GamerSam made Linux on phones possible which converts your phone into PC.
Desktop mode on phones are now reality and race is started.
Arm64 Linux on phones is future combined with Android via Samsung DEX is full desktop experience and is bombshell !
It can work without Samsung DEX, but is much easier that way !

Thank you for your build of LossLessCut for Arm64 Linux !

https://github.com/RandomCoderOrg/ubuntu-on-android/discussions/129

..

![image](https://user-images.githubusercontent.com/57914925/158499275-3ec26068-76db-484e-a0e9-4e56593b4a90.png)

..

**Comment by v8karlo** [comment]
Install Ubuntu on phone with termux-x11 fast GUI.

https://github.com/RandomCoderOrg/ubuntu-on-android/discussions/134

There are also instructions how to install LossLessCut on Phone Arm64 Ubuntu. 

Full desktop experience on mobile phone !

.
